src/ui/window.c

```diff
@@ -11,28 +11,47 @@
 #define DEFAULT_WIDTH 640
 #define DEFAULT_HEIGHT 480
 
-static void win_destroy_joystick(struct window *win) {
-  if (win->joystick) {
-    SDL_JoystickClose(win->joystick);
-    win->joystick = NULL;
+static void win_destroy_joystick(struct window *win, struct _SDL_Joystick *joystick) {
+  for(int i = 0; i < MAX_JOYSTICKS; i++) {
+    if(win->joysticks[i] == joystick) {
+      SDL_JoystickClose(win->joysticks[i]);
+      win->joysticks[i] = NULL;
+    }
   }
 }
 
 static void win_init_joystick(struct window *win) {
-  win_destroy_joystick(win);
+  /* connect joystick to first open slot */
 
-  /* open the first connected joystick */
-  for (int i = 0; i < SDL_NumJoysticks(); ++i) {
-    win->joystick = SDL_JoystickOpen(i);
+  for (int i = 0; i < MAX_JOYSTICKS; i++) {
 
-    if (win->joystick) {
-      LOG_INFO("Opened joystick %s (%d)", SDL_JoystickName(win->joystick), i);
-      break;
+    if(!win->joysticks[i]) {
+      win->joysticks[i] = SDL_JoystickOpen(i);
```

Comment:
Are the SDL device_index's guaranteed stable throughout joystick adding / removing?

If not, it seems this could _possibly_ cause the same physical joystick being assigned to multiple internal joysticks / prevent a joystick from being ever opened if it's assigned the index of a previously opened joystick.

We may want to call `win_init_joystick` on startup once in this form, and on subsequent `SDL_JOYDEVICEADDED` events call a new `win_create_joystick(struct window *win, int device_index)` which assigns the first non-NULL `win->joystick[]` the return value of `SDL_JoystickOpen(device_index)`